pycbc/neutron_stars/eos_utils.py

```diff
@@ -0,0 +1,208 @@
+    mass for a given neutron star mass
+
+    Parameters
+    ----------
+    ns_mass : float
+        The mass of the neutron star, in solar masses.
+    eos : str
+        Name of the equation of state.
+
+    Returns
+    -------
+    ns_compactness : float
+        Compactness parameter of the neutron star.
+    ns_b_mass : float
+        Baryonic mass of the neutron star.
+    """
+    if str(type(ns_mass)) == '<class \'numpy.ndarray\'>':
+        input_is_array = True
+    if eos in NS_SEQUENCES:
+        ns_seq, ns_max = load_ns_sequence(eos)
+        try:
+            if any(ns_mass > ns_max) and input_is_array:
+                raise ValueError(
+                    f'Maximum NS mass for {eos} is {ns_max}, received masses '
+                    f'up to {max(ns_mass[ns_mass > ns_max])}')
+        except TypeError:
+            if ns_mass > ns_max and not input_is_array:
+                raise ValueError(
+                    f'Maximum NS mass for {eos} is {ns_max}, received '
+                    f'{ns_mass}')
+        # Interpolate NS compactness and rest mass
+        ns_compactness = interp_grav_mass_to_compactness(ns_mass, ns_seq)
+        ns_b_mass = interp_grav_mass_to_baryon_mass(ns_mass, ns_seq)
+    elif eos in lalsim.SimNeutronStarEOSNames:
+        #eos_obj = lalsim.SimNeutronStarEOSByName(eos)
+        #eos_fam = lalsim.CreateSimNeutronStarFamily(eos_obj)
+        #r_ns = lalsim.SimNeutronStarRadius(ns_mass * lal.MSUN_SI, eos_obj)
+        #ns_compactness = lal.G_SI * ns_mass * lal.MSUN_SI / (r_ns * lal.C_SI**2)
+        raise NotImplementedError(
+            'LALSimulation EOS interface not yet implemented!')
+    else:
+        raise NotImplementedError(
+            f'{eos} is not implemented! Available are: '
+            f'{NS_SEQUENCES + list(lalsim.SimNeutronStarEOSNames)}')
+    return (ns_compactness, ns_b_mass)
+
+
+def foucart18(
+        eta, ns_compactness, ns_b_mass, bh_spin_mag, bh_spin_pol, interp=True):
+    """Function that determines the remnant disk mass of an NS-BH system
+    using the fit to numerical-relativity results discussed in
+    `Foucart, Hinderer & Nissanke, PRD 98, 081501(R) (2018)`_.
+
+    .. _Foucart, Hinderer & Nissanke, PRD 98, 081501(R) (2018):
+        https://doi.org/10.1103/PhysRevD.98.081501
+
+    Parameters
+    ----------
+    eta : float
+        The symmetric mass ratio of the system
+        (note: primary is assumed to be the BH).
+    ns_compactness : float
+        NS compactness parameter.
+    ns_b_mass : float
+        Baryonic mass of the NS.
+    bh_spin_mag: float
+        Dimensionless spin magnitude of the BH.
+    bh_spin_pol : float
+        The tilt angle of the BH spin.
+    interp: bool, optional
+        Whether to interpolate over a pre-computed grid of values of
+        the ISSO for systems with BH spin magnitude < 0.9. Otherwise
+        solve via the PG_ISSO_solver routine for all systems.
+        Default = True.
```

Comment:
Check this, it looks like some of those parameters can take Numpy arrays, not just float.